tests/transform/test_genes_biodomains.py

```diff
@@ -40,57 +43,50 @@ def test_count_grouped_total_two_groups(self):
-#         [{"biodomain": "c", "go_terms": ["c"]}, {"biodomain": "d", "go_terms": ["d"]}],
-#         [{"biodomain": "e", "go_terms": ["e"]}, {"biodomain": "f", "go_terms": ["f"]}],
-#     ]
-#     test_biodomains = transform.transform_biodomains(datasets=test_datasets)
-#     assert list(test_biodomains["gene_biodomains"]) == expected_gene_biodomains_col
-
-
-# df = pd.DataFrame(
-#     {'team id': [np.nan, 0, 1, 2],
-#      'team.Name': ['MSN', 'Team 1', 'Team 2', np.nan],
-#      'team-Sco@#&': ['x', 'y', 'z', "na"]})
-
-# def test_standardize_column_names():
-
-#     result_df = transform.standardize_column_names(df)
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert list(result_df.columns) == ['team_id', 'team_name', 'team-sco']
-
-
-# def test_standardize_values():
-
-#     assert df.isna().sum().sum() == 2
-
-#     result_df = transform.standardize_values(df)
-
-#     assert type(result_df) is pd.core.frame.DataFrame
-#     assert result_df.isna().sum().sum() == 0
-#     assert result_df.shape == (4, 3)
-
-# def test_rename_columns():
-#     refresh_df = pd.DataFrame(
-#         {'team id': [np.nan, 0, 1, 2],
-#          'team.Name': ['MSN', 'Team 1', 'Team 2', np.nan],
-#          'team-Sco@#&': ['x', 'y', 'z', "na"]})
-
-#     bad_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&"})
-#     assert type(bad_result_df) is pd.core.frame.DataFrame
-#     assert list(bad_result_df.columns) == ["team id", "team.Name", "team-Sco@#&"]
-
-#     partial_good_result_df = transform.rename_columns(df=refresh_df, column_map={"team-Sco@#&": "team_scope"})
-#     assert list(partial_good_result_df.columns) == ['team id', 'team.Name', 'team_scope']
-#     assert type(partial_good_result_df) is pd.core.frame.DataFrame
+class TestTransformGenesBiodomains:
+    data_files_path = "tests/test_assets/data_files"
+    pass_test_data = [
+        (  # pass with good data
+            "biodomains_test_input.csv",
+            "genes_biodomains.json",
+        ),
+        (  # pass with imperfect data
+            "biodomains_test_input_bad_but_should_pass.csv",
+            "genes_biodomains_bad_output_but_should_pass.json",
+        ),
+    ]
+    pass_test_ids = [
+        "Pass with good data",
+        "Pass with imperfect data",
+    ]
+    fail_test_data = [
+        "biodomains_test_input_bad_should_fail.csv",  # fail with bad data
+    ]
+    fail_test_ids = [
+        "Fail with bad data",
+    ]
+
+    @pytest.mark.parametrize(
+        "input_file, expected_output_file", pass_test_data, ids=pass_test_ids
+    )
+    def test_transform_genes_biodomains_should_pass(
+        self, input_file, expected_output_file
+    ):
+        input_df = pd.read_csv(os.path.join(self.data_files_path, "input", input_file))
+        output_df = genes_biodomains.transform_genes_biodomains(
+            datasets={"genes_biodomains": input_df}
+        )
+        expected_df = pd.read_json(
```

Comment:
One issue with using the JSON output instead of the csv output I provided, which I didn't think of until now: In between running the transform and writing to JSON in `process.py`, we have an `agora_rename` option to rename columns _after_ the transformation. This doesn't affect `genes_biodomains`, but does affect some of the older transforms which output different column names than the agora app expects. So for those transforms, using the JSON for comparison will fail because the column names are different, if that makes sense. It might be good to default to using the output csv (which I wrote immediately after the transform and before `agora_rename`) even for transforms that don't have this issue. 

Otherwise, this looks great and is so much cleaner than what we had!

Reply:
So the CSV file is the "true" output of `transform_genes_biodomains`, whereas the JSON file is the output of the entire processing run?

Reply:
Yeah that's correct, although I think I need to change that. It just occurred to me that we have nested fields in some transforms (probably this one too) that don't translate to csv well. Let me think for a while.

Reply:
Yeah, I had noticed this. I think the simplest solution is to export the output of each transform function (before any renaming may take place) as JSON instead of CSV. That way when you import that file to use in a test, the data frame matches the one produced by running the transform function.

Reply:
Perfect, thanks!